third-party/crowdsec/src/test/java/org/apache/james/CrowdsecExtension.java

```diff
@@ -23,59 +23,39 @@
 import java.util.UUID;
 
 import org.apache.james.util.docker.RateLimiters;
-import org.junit.jupiter.api.extension.BeforeAllCallback;
 import org.junit.jupiter.api.extension.ExtensionContext;
 import org.testcontainers.containers.GenericContainer;
 import org.testcontainers.containers.wait.strategy.HostPortWaitStrategy;
 
-public class CrowdsecExtension implements BeforeAllCallback {
+public class CrowdsecExtension implements GuiceModuleTestExtension {
     public static final Duration STARTUP_TIMEOUT = Duration.ofMinutes(5);
     public static final int CROWDSEC_PORT = 8080;
     public static final String CROWDSEC_IMAGE = "crowdsecurity/crowdsec:v1.5.4";
-    private static final GenericContainer<?> crowdsecContainer = new GenericContainer<>(CROWDSEC_IMAGE)
-        .withCreateContainerCmdModifier(cmd -> cmd.withName("james-crowdsec-test-" + UUID.randomUUID()))
-        .withExposedPorts(CROWDSEC_PORT)
-        .withStartupTimeout(STARTUP_TIMEOUT)
-        .waitingFor(new HostPortWaitStrategy().withRateLimiter(RateLimiters.TWENTIES_PER_SECOND));
     public static final String API_KEY_BOUNCER = "myKey";
 
-    static {
-        if (!crowdsecContainer.isRunning()) {
-            crowdsecContainer.start();
-        }
-    }
+    private final GenericContainer<?> crowdsecContainer;
 
-    private Crowdsec crowdsec;
+    public CrowdsecExtension() {
+        this.crowdsecContainer = new GenericContainer<>(CROWDSEC_IMAGE)
+            .withCreateContainerCmdModifier(cmd -> cmd.withName("james-crowdsec-test-" + UUID.randomUUID()))
+            .withExposedPorts(CROWDSEC_PORT)
+            .withStartupTimeout(STARTUP_TIMEOUT)
+            .waitingFor(new HostPortWaitStrategy().withRateLimiter(RateLimiters.TWENTIES_PER_SECOND));;
+    }
 
     @Override
     public void beforeAll(ExtensionContext context) {
-        crowdsec = new Crowdsec(crowdsecContainer);
+        if (!crowdsecContainer.isRunning()) {
+            crowdsecContainer.start();
+        }
     }
 
-    public Crowdsec getCrowdsec() {
-        return crowdsec;
+    @Override
+    public void afterAll(ExtensionContext extensionContext) {
+        crowdsecContainer.stop();
```

Comment:
This would prevent the CrowdSec container from being re-used across test classes.

No need to stop the container here IMO.

rf: https://github.com/apache/james-project/pull/1772#issuecomment-1790312565

Reply:
Updated